rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize.cpp

```diff
@@ -54,10 +53,82 @@ void ResizeNode::create_node()
     vx_status status;
     if((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
         THROW("Adding the resize (vxExtrppNode_ResizebatchPD) node failed: "+ TOSTR(status))
+}
 
+void ResizeNode::update_node() {
+    std::vector<uint32_t> src_h_dims, src_w_dims;
+    src_w_dims = _inputs[0]->info().get_roi_width_vec();
+    src_h_dims = _inputs[0]->info().get_roi_height_vec();
+    for (unsigned i = 0; i < _batch_size; i++) {
+        _src_width = src_w_dims[i];
+        _src_height = src_h_dims[i];
+        _dst_width = _out_width;
+        _dst_height = _out_height;
+        adjust_out_roi_size();
+        _dst_width = std::min(_dst_width, _outputs[0]->info().width());
+        _dst_height = std::min(_dst_height, _outputs[0]->info().height_single());
+        _dst_roi_width_vec.push_back(_dst_width);
+        _dst_roi_height_vec.push_back(_dst_height);
+    }
+    vx_status width_status, height_status;
+    width_status = vxCopyArrayRange((vx_array)_dst_roi_width, 0, _batch_size, sizeof(vx_uint32), _dst_roi_width_vec.data(), VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    height_status = vxCopyArrayRange((vx_array)_dst_roi_height, 0, _batch_size, sizeof(vx_uint32), _dst_roi_height_vec.data(), VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST);
+    if(width_status != 0 || height_status != 0)
+        WRN("ERROR: vxCopyArrayRange _dst_roi_width or _dst_roi_height failed " + TOSTR(width_status) + "  " + TOSTR(height_status));
+    _outputs[0]->update_image_roi(_dst_roi_width_vec, _dst_roi_height_vec);
+    _dst_roi_width_vec.clear();
+    _dst_roi_height_vec.clear();
 }
 
-void ResizeNode::update_node()
-{
-  
-}
\ No newline at end of file
+void ResizeNode::init(unsigned dest_width, unsigned dest_height, RocalResizeScalingMode scaling_mode,
+                      std::vector<unsigned> max_size, RocalResizeInterpolationType interpolation_type) {
+    _scaling_mode = scaling_mode;
+    _out_width = dest_width;
+    _out_height = dest_height;
+    if(max_size.size() > 0) {
```

Comment:
what is the value of max_width and max_height for else clause?

Reply:
Only if user specifies the value the max_width and max_height it will be set, Otherwise it would be zero and not taken into consideration

Reply:
If max_size is a read_only parameter, please pass by reference with const type

Reply:
Done